phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -1993,19 +1992,31 @@ public void clearBatch() throws SQLException {
     @Override
     public int[] executeBatch() throws SQLException {
         int i = 0;
+        int[] returnCodes = new int [batch.size()];
+        Arrays.fill(returnCodes, -1);
+        boolean autoCommit = connection.getAutoCommit();
+        connection.setAutoCommit(false);
```

Comment:
we need to restore the state of auto commit setting before we return from this function

Reply:
Good catch! I saved the value but forgot to use it.